simdjson-flags.cmake

```diff
@@ -75,22 +76,22 @@ if(SIMDJSON_ENABLE_THREADS)
   find_package(Threads REQUIRED)
   set(CMAKE_THREAD_PREFER_PTHREAD TRUE)
   set(THREADS_PREFER_PTHREAD_FLAG TRUE)
-  target_link_libraries(simdjson-flags INTERFACE ${CMAKE_THREAD_LIBS_INIT} )
+  target_link_libraries(simdjson-public-flags INTERFACE ${CMAKE_THREAD_LIBS_INIT} )
 endif()
 
 option(SIMDJSON_SANITIZE "Sanitize addresses" OFF)
 if(SIMDJSON_SANITIZE)
-  target_compile_options(simdjson-flags INTERFACE -fsanitize=address -fno-omit-frame-pointer -fsanitize=undefined -fno-sanitize-recover=all)
-  target_link_libraries(simdjson-flags INTERFACE -fsanitize=address -fno-omit-frame-pointer -fsanitize=undefined -fno-sanitize-recover=all)
+  target_compile_options(simdjson-public-flags INTERFACE -fsanitize=address -fno-omit-frame-pointer -fsanitize=undefined -fno-sanitize-recover=all)
+  target_link_libraries(simdjson-public-flags INTERFACE -fsanitize=address -fno-omit-frame-pointer -fsanitize=undefined -fno-sanitize-recover=all)
 
   # Ubuntu bug for GCC 5.0+ (safe for all versions)
   if (CMAKE_COMPILER_IS_GNUCC)
-    target_link_libraries(simdjson-flags INTERFACE -fuse-ld=gold)
+    target_link_libraries(simdjson-public-flags INTERFACE "-fuse-ld=gold")
```

Comment:
I have no opposition to the addition of quotes in general (I have no skin in the game either way), but why do this here and not other places?

Reply:
Probably not necessary, anymore.
 I had an error before saying that parameter -ld not recognized, and thought maybe quoting fixes but that wasn't the issue if I remember correcetly.
I'll verify and remove the quotes for consistency